File: src/za/co/adhd_developers/questions/Question29.java
package za.co.adhd_developers.questions;

import za.co.adhd_developers.tools.Question;

import java.util.ArrayList;

public class Question29 implements Question
{
    private int lowerLimit = 2;
    private int upperlimit = 100;

    private int answer = 0;

    private ArrayList<Double> terms = new ArrayList<>();

    @Override
    public void doWork()
    {
        int a = this.lowerLimit;

        while (a <= this.upperlimit)
        {
            int b = this.lowerLimit;
            while (b <= this.upperlimit)
            {
                double result = Math.pow(a,b);

                if (!this.terms.contains(result))
                {
                    this.terms.add(result);
                }
                b++;
            }
            a++;
        }

        this.answer = this.terms.size();
    }

    @Override
    public void printAnswer()
    {
        System.out.println("=================================");
        System.out.println("Question 29");
        System.out.println("Amount of distinct terms in the sequence, is: "+this.answer);
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question29.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question

class Question29 :Question {

    val limit = 101
    var answer = 0
    override fun doWork() {

        var terms: MutableSet<Double> = mutableSetOf()

        for (a in 2 until limit) {
            for (b in 2 until limit) {
                terms.add(Math.pow(a.toDouble(),b.toDouble()))
            }
        }
        answer = terms.size
    }

    override fun printAnswer() {
        println("=================================")
        println("Question 29")
        println("Amount of distinct terms in the sequence, is: " + this.answer)
        println("=================================")
    }
}